show error message in event tooltip

From: Ondrej Jirman <ondrej.jirman@zonio.net>

calendar/ChangeLog:

2008-04-06  Ondrej Jirman  <megous@megous.com>

	** Fix for bug #not yet submitted

	* gui/e-calendar-view.c: (e_calendar_view_get_tooltips):
	Display error string in event's tooltip from X-EVOLUTION-ERROR event
	property.
---

 calendar/gui/e-calendar-view.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)


diff --git a/calendar/gui/e-calendar-view.c b/calendar/gui/e-calendar-view.c
index 62f929b..8a4542b 100644
--- a/calendar/gui/e-calendar-view.c
+++ b/calendar/gui/e-calendar-view.c
@@ -2196,6 +2196,20 @@ e_calendar_view_get_tooltips (ECalendarViewEventData *data)
 
 	gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
 	g_free (tmp);
+	
+	/* display error */
+	e_cal_component_get_x_property (newcomp, "X-EVOLUTION-ERROR", &str);
+	if (str) {
+		tmp = g_strdup_printf (_("Error: %s"), str);
+		label = gtk_label_new (NULL);
+		gtk_label_set_markup ((GtkLabel *)label, tmp);
+		hbox = gtk_hbox_new (FALSE, 0);
+		gtk_box_pack_start ((GtkBox *)hbox, label, FALSE, FALSE, 0);
+		ebox = gtk_event_box_new ();
+		gtk_container_add ((GtkContainer *)ebox, hbox);
+		gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
+		g_free (tmp);
+	}
 
 	e_cal_component_get_organizer (newcomp, &organiser);
 	if (organiser.cn) {
